api/src/pcapi/core/finance/api.py

```diff
@@ -835,6 +839,38 @@ def _write_csv(
     return path
 
 
+def _append_csv(
+    path: pathlib.Path,
+    rows: typing.Iterable = None,
+    batched_rows: typing.Iterable = None,
+    row_formatter: typing.Callable[typing.Any, typing.Iterable] = lambda row: row,  # type: ignore [misc]
+    compress: bool = False,
+) -> pathlib.Path:
+    assert (rows is not None) ^ (batched_rows is not None)
```

Comment:
Pourquoi une syntaxe avec assert plutôt que de raise explicitement une erreur métier ?

Reply:
C'est moi qui ai écrit ça. J'utilise parfois `assert` dans ce genre de cas où l'on veut vérifier qu'on appelle pas n'importe comment une fonction. Mais c'est probablement une mauvaise habitude. Je pourrais sûrement réécrire comme ça : 

```suggestion
    if not ((rows is not None) ^ (batched_rows is not None)):
        raise RuntimeError("_write_csv must be called with 'rows' or 'batched_rows', not both.")
```